fairscale/nn/pipe/balance/__init__.py

```diff
@@ -23,13 +23,20 @@
 
     import torch
     from fairscale.nn import Pipe
-    from fairscale.nn.balance import balance_by_time
+    from fairscale.nn.pipe.balance import balance_by_time
 
     sample = torch.empty(128, 3, 224, 224)
     balance = balance_by_time(torch.cuda.device_count(), model, sample)
 
     pipe = Pipe(model, balance, chunks=8)
 
+.. note::
+    balance_by_time does not work with inplace ReLU because we exhausetively search
+    every partition boundary, which could hit an inplace ReLU.
+
+.. note::
+    If the model is larger than a single CUDA device memory, use "cpu"
+    in the balance_by_time function.
```

Comment:
Just a note, when I was using it on the 10B model, it crashed with a segfault. Obviously I had to use the cpu device. So I didn't get a useful split out of this function. In the end, my pipeline split has to be done by hand tuning anyway. :-(